internal/auth/token.go

```diff
@@ -0,0 +1,53 @@
+package auth
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+)
+
+func GetUserIDFromToken(c *gin.Context) (uuid.UUID, error) {
```

Comment:
This function is used in every handler you have created. Can you think of every other way you could have used this in the handlers ? Through gin maybe ?

Reply:
I should've probably added the code to the gin pipeline e.g. `router.use(someJwtCheckFunction)` in `main.go`.
I could abort the request there if a valid token is absent, and store the userId extracted from a valid token in the gin context.

Reply:
Yes exactly, adding this as a middleware is what I was thinking too.